### PR TITLE
Initial Address book support

### PR DESCRIPTION
**Pull request by Kaian**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [X] Upport from existing Pull request #1984

#### Description
Add new business table to store Contacts information.
Contacts for existing vPBX users will be added and maintained automatically.

Current contacts fields:
 - Name
 - Lastname
 - Email
 - Work Phone
 - Mobile Phone
 - Other Phone

Currently Contacts are not being used in any logic but can be created and retrieved through Rest API.

### CODE REVIEW

File: web/portal/client/src/entities/Contact/Contact.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+export const ChildDecorator: ChildDecoratorType = (props) => {
+  const { routeMapItem, row } = props;
+
+  if (isEntityItem(routeMapItem) && routeMapItem.entity.iden === Contact.iden) {
+    const isDeletePath = routeMapItem.route === `${Contact.path}/:id`;
+    const allowDelete = row.user === null;
+    console.log(row);
+    console.log(isDeletePath);
+    console.log(allowDelete);
```
Comment (mmadariaga) [inline web/portal/client/src/entities/Contact/Contact.tsx:67]: Remove console logs

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: web/portal/client/src/entities/Contact/Contact.tsx (hunk 1 above)
Comment (Kaian) [reply web/portal/client/src/entities/Contact/Contact.tsx:67]: Good catch! :+1:

File: schema/DoctrineMigrations/Version20230113114920.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+namespace Application\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20230113114920 extends AbstractMigration
```
[the remark this answers is not in the document]
Comment (Kaian) [reply schema/DoctrineMigrations/Version20230113114920.php]: Fixed!

**Comment by Kaian** [comment]
Updated with requested changes.
Also added:

- Changed API spec to add delete restrictions on user contacts (those with userId != NULL)
- Updated DTO denormalize to ignore some field updates in user contacts 


Please, review those extra changes too :pray: 

Thanks!

**Review by mmadariaga — APPROVED** [review]

**Comment by Kaian** [comment]
Added this fixture of a contact from the second company:

https://github.com/irontec/ivozprovider/blob/77a913b9be870b2329044573fec261c7a10981f2/library/DataFixtures/ORM/ProviderContact.php#L74-L88

And this test:

https://github.com/irontec/ivozprovider/blob/bb9e15434890714553634c70ba5e1f217b6a32d1/web/rest/client/features/provider/contact/removeContact.feature#L14-L19

**Comment by mmadariaga** [comment]
LGTM

**Pull request merged**